frontend/packages/console-shared/src/components/spotlight/InteractiveSpotlight.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { PopperOptions } from 'popper.js';
+import { Popper } from '../popper';
+import './spotlight.scss';
+
+type InteractiveSpotlightProps = {
+  element: Element;
+};
+
+const InteractiveSpotlight: React.FC<InteractiveSpotlightProps> = ({ element }) => {
+  const { height, width } = element.getBoundingClientRect();
+  const style: React.CSSProperties = {
+    height,
+    width,
+  };
+  const popperOptions: PopperOptions = {
+    modifiers: {
+      preventOverflow: {
+        enabled: false,
+      },
+    },
+  };
+  const isInViewport = (elementToCheck: Element) => {
+    const rect = elementToCheck.getBoundingClientRect();
+    return (
+      rect.top >= 0 &&
+      rect.left >= 0 &&
+      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+    );
+  };
```

Comment:
This is a utility function. Move it out of the render closure.

Reply:
done